enumerable.rb

```diff
@@ -0,0 +1,115 @@
+module Enumerable
+  def my_each
+    to_a.length.times { |i| yield self[i] }
+    self
+  end
+
+  def my_each_with_index
+    to_a.my_each { |each| yield each, index(each) }
+    self
+  end
+
+  def my_select
+    array = []
+    to_a.my_each { |each| array.push(each) if yield each }
+    array
+  end
```

Comment:
## my_select

- [ ] returns an enumerator if no block is given
```ruby
array.my_select
```
Expected: Enumerator
got: LocalJumpError:
        no block given (yield)